### DISCUSSION TITLE
Cron Functionality - Pg-moneta

### DISCUSSION THREAD

**Discussion by Anshgupta0123** [pgmoneta/pgmoneta#312] [Ideas]
Objective
Our goal is to integrate automated cron-based scheduling into pgmoneta to streamline PostgreSQL backups, ensuring they are consistent, reliable, and efficient.

Develop pgmoneta_cron Module
This involves creating a new module (pgmoneta_cron) specifically for handling the scheduling and execution of backup tasks using cron jobs. Cron jobs are time-based job schedulers in Unix-like operating systems.

Cron is a time-based job scheduler in Unix-like operating systems, including Linux and macOS. It allows users to schedule tasks or commands to run periodically at specified intervals or times.

Here's how it works:

Cron Schedule: Users define a schedule using a special syntax in a file called a cron table (crontab). The syntax includes fields for specifying minutes, hours, days of the month, months, and days of the week when the task should run. For example, a cron schedule might specify that a task should run every day at 3:00 AM.

Commands or Scripts: Users associate commands or scripts with the cron schedule. These commands or scripts are the tasks that will be executed according to the schedule. They can be anything from simple shell commands to complex scripts or programs.

Execution: The cron daemon (cron.d) continuously checks the cron table for scheduled tasks. When it finds a task whose schedule matches the current time, it executes the associated command or script.

Output and Logging: Any output generated by the command or script, including errors or messages, is typically sent via email to the user who owns the cron job. Additionally, system logs may record information about cron job execution for troubleshooting and monitoring purposes.

**Comment by jesperpedersen** [comment]
I don't know what you mean about module, but yes, having a cron like syntax in the configuration would be a good thing in order to make sure that f.ex. backups are run at that time.

Since we don't have email integration, just do logging for now

**Comment by jesperpedersen** [comment]
The important part is to get a work branch up and running such that we can provide feedback on it

**Comment by Anshgupta0123** [comment]
By module I want to say that lets have a this functionality to pg-moneta as soon as possible as the working branch . 
For this i wanna know should we proceed with it in the most efficient .

**Reply by jesperpedersen** [reply]
I don't know what you mean, but you should have a `cron` for the general configuration, and then one per server, like

```
[pgmoneta]
...
cron = 0 * * * *

[primary]
cron = 0 * * * *
```

When you have your working branch then you can post it here, and we can provide feedback before you submit your pull request

**Comment by Anshgupta0123** [comment]
Got it .I'll start working on it .

**Comment by kbzowski** [comment]
What's wrong with running pgmonet-cli as a cron task?

**Reply by jesperpedersen** [reply]
There is nothing wrong with it - it is a matter of having the option, if you don't want to depend on `crontab`
